hogben/tests/test_samples.py

```diff
@@ -0,0 +1,313 @@
+    g = fisher(qs, sample.params, counts, models)
+
+    np.testing.assert_allclose(g, angle_info, rtol=1e-08)
+
+@patch('hogben.models.samples.Sample._get_sld_profile')
+@patch('hogben.models.samples.save_plot', side_effect=mock_save_plot)
+def test_sld_profile_valid_figure(_mock_save_plot,
+                                  mock_sld_profile, refnx_sample):
+    """
+    Tests whether the sld_profile function succesfully outputs a figure
+    """
+    mock_sld_profile.return_value = ([0, 10, 60, 110, 160, 210],
+                                     [4, 9, -2, 9, -2, 9])
+
+
+    # Use temporary directory, so it doesn't leave any files after testing
+    with tempfile.TemporaryDirectory() as temp_dir:
+        refnx_sample.sld_profile(temp_dir)
+        img_test = os.path.join(temp_dir, refnx_sample.name, 'sld_profile.png')
+        assert os.path.isfile(img_test)
+
+@patch('hogben.models.samples.Sample._get_reflectivity_profile')
+@patch('hogben.models.samples.save_plot', side_effect=mock_save_plot)
+def test_reflectivity_profile_valid_figure(_mock_save_plot,
+                                           _mock_reflectivity_profile,
+                                           refnx_sample):
+    """
+    Tests whether the reflectivity_profile function succesfully outputs a
+    figure
+    """
+    _mock_reflectivity_profile.return_value = ([0, 0.05, 0.1, 0.15, 0.2],
+                                               [1, 0.9, 0.8, 0.75, 0.8])
+    # Use temporary directory, so it doesn't leave any files after testing
+    with tempfile.TemporaryDirectory() as temp_dir:
+        refnx_sample.reflectivity_profile(temp_dir)
+        img_test = os.path.join(temp_dir, refnx_sample.name,
+                                'reflectivity_profile.png')
+        assert os.path.isfile(img_test)
+
+@patch('hogben.models.samples.save_plot', side_effect=mock_save_plot)
+@pytest.mark.parametrize('sample_class', ("refnx_sample",
+                                         "refl1d_sample"))
+def test_sld_profile_length(_mock_save_plot, sample_class,
+                                        request):
+    """
+    Tests whether _get_sld_profile() succesfully retrieves two arrays with
+    equal lengths, representing an SLD profile that can be plotted in a figure
+    """
+    sample = request.getfixturevalue(sample_class)
+    z, slds = sample._get_sld_profile()
+    assert len(z) == len(slds)
+    assert len(z) > 0 # Make sure arrays are not empty
+
+@pytest.mark.parametrize('sample_class', ("refnx_sample",
+                                         "refl1d_sample"))
+def test_reflectivity_profile_positive(sample_class,
+                                        request):
+    """
+    Tests whether _get_reflectivity_profile() succesfully obtains reflectivity
+    values that are all positively valued
+    """
+    sample = request.getfixturevalue(sample_class)
+    q, r = sample._get_reflectivity_profile(0.005, 0.4, 500, 1, 1e-7, 2)
+    assert min(r) > 0
```

Comment:
I think that being less clever here is probably simplest:
`assert np.all(np.greater(r, 0.0))`